From 40a8f62d4aa3fd1d31f88121240f2abfb44c766c Mon Sep 17 00:00:00 2001
From: Juan Quintela <quintela@redhat.com>
Date: Wed, 19 Oct 2011 23:47:42 +0200
Subject: [PATCH 07/18] buffered_file: Use right "opaque"

RH-Author: Juan Quintela <quintela@redhat.com>
Message-id: <01455989d4bf3b19468eafe7fe4c770df1bbdcfb.1319066771.git.quintela@redhat.com>
Patchwork-id: 34434
O-Subject: [PATCH qemu-kvm RHEL-6.2 07/16] buffered_file: Use right "opaque"
Bugzilla: 669581
RH-Acked-by: Amit Shah <amit.shah@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Orit Wasserman <owasserm@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>

buffered_close 's' variable is of type QEMUFileBuffered, and
wait_for_unfreeze() expect to receive a MigrationState, that
'coincidentaly' is s->opaque.

Signed-off-by: Juan Quintela <quintela@redhat.com>
Reviewed-by: Anthony Liguori <aliguori@us.ibm.com>
---
 buffered_file.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 buffered_file.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/buffered_file.c b/buffered_file.c
index 4731623..d62bc37 100644
--- a/buffered_file.c
+++ b/buffered_file.c
@@ -177,7 +177,7 @@ static int buffered_close(void *opaque)
     while (!s->has_error && s->buffer_size) {
         buffered_flush(s);
         if (s->freeze_output)
-            s->wait_for_unfreeze(s);
+            s->wait_for_unfreeze(s->opaque);
     }
 
     ret = s->close(s->opaque);
-- 
1.7.4.4

